### PR TITLE
Return models after CV

### LINKED ISSUE #1524: Return fitted models after cv
Hi!

There is no opportunity to get fitted models after cv, because catboost.cv returns only evaluation metric scores. At the same time popular ml libraries have such option in some kind.

For LightGBM there is optional argument [`return_cvbooster`](https://lightgbm.readthedocs.io/en/latest/pythonapi/lightgbm.cv.html):

```python
cv = lgb.cv(params, X_train, show_stdv=False, stratified=True, verbose_eval=True,
            num_boost_round=trees, nfold=5, early_stopping_rounds=50,
            return_cvbooster=True, seed=72, )
```

XGBoost support such opportunity by using callbacks:

```python
cvbooster = []

def save_best_model(env):
    global cvbooster
    cvbooster[:] = [cvpack.bst for cvpack in env.cvfolds]

cv = xgb.cv(params, X_train, stratified=True, verbose_eval=True,
            num_boost_round=trees, nfold=5, early_stopping_rounds=50,
            seed=72, callbacks=[save_best_model, ])
```

Final models fitted on cv can be used in bagging ensemble. Also having individual models for each split will help investigating problems in data on specific fold.

It will be awesome to get such feature in future.

### PR DESCRIPTION
**Pull request by vklyukin**
Closes #1524

I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=ru.

### CODE REVIEW

**Comment by Evgueni-Petrov-aka-espetrov** [comment]
@arcadia-devtools Ship it!

**Comment by arcadia-devtools** [comment]
CLA already signed

**Comment by arcadia-devtools** [comment]
@Evgueni-Petrov-aka-espetrov, internal review request created: [1729111](https://a.yandex-team.ru/review/1729111)

**Comment by arcadia-devtools** [comment]
Check [1729111](https://a.yandex-team.ru/review/1729111) failed: 5 test(s) broken

File: catboost/python-package/ut/medium/test.py [hunk 1]
```diff
@@ -2930,6 +2930,54 @@ def test_cv_with_save_snapshot(task_type):
...
+        assert np.allclose(cv_model_roc_auc, single_model_roc_auc, atol=results["test-AUC-std"][0])
+
+        # Other methods
+        model.calc_leaf_indexes(test_pool)
+        try:
+            model.compare(single_model, test_pool, ["AUC"])
+        except ImportError as ie:
+            pytest.xfail(str(ie)) if str(ie) == "No module named widget" \
+                else pytest.fail(str(ie))
+        assert model.get_all_params().keys() == single_model.get_all_params().keys()
```
Comment (Evgueni-Petrov-aka-espetrov) [inline catboost/python-package/ut/medium/test.py:2974]: this assert fails on fresh master as follows -- please fix
```python
catboost/python-package/ut/medium/test.py:2974: in test_cv_return_models
    assert model.get_all_params().keys() == single_model.get_all_params().keys()
E   AssertionError: assert ['bayesian_ma...s_count', ...] == ['bayesian_mat...s_count', ...]
E     Right contains more items, first extra item: 'class_names'
E     Full diff:
E     [u'bayesian_matrix_reg',
E     u'leaf_estimation_backtracking',
E     u'has_time',
E     u'min_data_in_leaf',
E     u'model_shrink_rate',
E     u'classes_count',
E     u'max_leaves',
E     u'counter_calc_method',
E     u'iterations',
E     u'max_ctr_complexity',
E     u'combinations_ctr',
E     u'sparse_features_conflict_fraction',
E     u'boosting_type',
E     u'task_type',
E     u'pool_metainfo_options',
E     u'posterior_sampling',
E     u'sampling_frequency',
E     u'loss_function',
E     u'l2_leaf_reg',
E     u'border_count',
E     u'ctr_leaf_count_limit',
E     u'bootstrap_type',
E     u'auto_class_weights',
E     u'rsm',
E     u'use_best_model',
E     u'leaf_estimation_iterations',
E     u'random_seed',
E     u'eval_metric',
E     u'permutation_count',
E     u'learning_rate',
E     u'nan_mode',
E     u'penalties_coefficient',
E     u'simple_ctr',
E     u'fold_permutation_block',
E     u'one_hot_max_size',
E     u'best_model_min_trees',
E     u'model_shrink_mode',
E     u'score_function',
E     u'store_all_simple_ctr',
E     u'boost_from_average',
E     u'model_size_reg',
E     u'grow_policy',
E     u'ctr_target_border_count',
E     u'random_strength',
E     u'subsample',
E     u'depth',
E     u'feature_border_type',
E     -  u'leaf_estimation_method']
E     ?                           ^
E     +  u'leaf_estimation_method',
E     ?                           ^
E     +  u'class_names']
```

**Review by Evgueni-Petrov-aka-espetrov — CHANGES_REQUESTED** [review]

**Review by Evgueni-Petrov-aka-espetrov — APPROVED** [review]

**Comment by Evgueni-Petrov-aka-espetrov** [comment]
@arcadia-devtools Ship it!

**Comment by arcadia-devtools** [comment]
@Evgueni-Petrov-aka-espetrov, internal review request updated: [1729111](https://a.yandex-team.ru/review/1729111)

**Comment by arcadia-devtools** [comment]
@Evgueni-Petrov-aka-espetrov, internal review request updated: [1729111](https://a.yandex-team.ru/review/1729111)

**Comment by arcadia-devtools** [comment]
Check [1729111](https://a.yandex-team.ru/review/1729111) passed

**Pull request merged**